Source/Core/VideoCommon/Assets/ShaderAsset.cpp

```diff
@@ -45,28 +49,38 @@ bool ParseShaderProperties(const VideoCommon::CustomAssetLibrary::AssetID& asset
     std::transform(type.begin(), type.end(), type.begin(),
                    [](unsigned char c) { return std::tolower(c); });
 
-    if (type == "sampler2d")
-    {
-      property.m_type = ShaderProperty::Type::Type_Sampler2D;
-    }
-    else if (type == "samplercube")
-    {
-      property.m_type = ShaderProperty::Type::Type_SamplerCube;
-    }
-    else if (type == "samplerarrayshared_main")
-    {
-      property.m_type = ShaderProperty::Type::Type_SamplerArrayShared_Main;
-    }
-    else if (type == "samplerarrayshared_additional")
+    static constexpr std::array<std::pair<std::string_view, ShaderProperty::Type>,
+                                static_cast<int>(ShaderProperty::Type::Type_Max)>
+        pairs = {{
+            {"sampler2d", ShaderProperty::Type::Type_Sampler2D},
+            {"samplercube", ShaderProperty::Type::Type_SamplerCube},
+            {"samplerarrayshared_main", ShaderProperty::Type::Type_SamplerArrayShared_Main},
+            {"samplerarrayshared_additional",
+             ShaderProperty::Type::Type_SamplerArrayShared_Additional},
+            {"int", ShaderProperty::Type::Type_Int},
+            {"int2", ShaderProperty::Type::Type_Int2},
+            {"int3", ShaderProperty::Type::Type_Int3},
+            {"int4", ShaderProperty::Type::Type_Int4},
+            {"float", ShaderProperty::Type::Type_Float},
+            {"float2", ShaderProperty::Type::Type_Float2},
+            {"float3", ShaderProperty::Type::Type_Float3},
+            {"float4", ShaderProperty::Type::Type_Float4},
+            {"rgb", ShaderProperty::Type::Type_RGB},
+            {"rgba", ShaderProperty::Type::Type_RGBA},
+            {"bool", ShaderProperty::Type::Type_Bool},
+        }};
+    if (const auto it = std::find_if(pairs.begin(), pairs.end(),
```

Comment:
Needs `#include <algorithm>`. It seems to be indirectly included anyway but yeah...

Reply:
(so does the `std::transform` that's already here, actually...)

Reply:
Good catch.  I need to make all these changes in material too.